scripts/new_upstream_snapshot.py

```diff
@@ -0,0 +1,627 @@
+    us what to do, we have to ask.
+    """
+    is_devel = is_first_devel_upload = known_first_devel_upload
+    is_first_sru = known_first_sru
+    if is_first_devel_upload and is_first_sru:
+        raise RuntimeError(
+            "Can't simultaneously be first SRU and first devel upload"
+        )
+    if (
+        not known_first_devel_upload
+        and not known_first_sru
+        and "~" not in env.changelog_details.version
+    ):
+        is_devel = True
+        if env.devel_distro != env.changelog_distro:
+            # Changelog shows devel release numbers, but not the current devel
+            # release. We need to know whether this is a new devel upload
+            # or the first SRU upload for a series.
+            print(
+                f"d/changelog shows current devel distro as "
+                f"{env.changelog_distro}, yet `distro-info` says it is "
+                f"{env.devel_distro}."
+            )
+            if (
+                input(
+                    f"Is this the first devel upload for {env.devel_distro} "
+                    "(y/N)? "
+                ).lower()
+                == "y"
+            ):
+                is_devel = True
+                is_first_devel_upload = True
+            elif (
+                input(
+                    f"Is this the first SRU for series {env.changelog_distro} "
+                    "(y/N)? "
+                ).lower()
+                == "y"
+            ):
+                is_devel = False
+                is_first_sru = True
+    return is_devel, is_first_devel_upload, is_first_sru
+
+
+def get_sru_bug(bug, no_sru_bug):
+    """Ask the user for an SRU bug if they did not provide one.
+
+    Don't ask if they tell us not to ask.
+    """
+    if not bug and not no_sru_bug:
+        bug = input(
+            "No SRU bug. Enter one now or leave blank for no SRU bug: "
+        ).strip()
+        if not bug:
+            bug = None
+    return bug
+
+
+def new_upstream_snapshot(
+    commitish: str,
+    bug: Optional[str] = None,
+    known_first_devel_upload: bool = False,
+    no_sru_bug: bool = False,
+    known_first_sru: bool = False,
+    post_patch: bool = False,
+) -> None:
+    """Perform a new upstream snapshot.
+
+    At a high level, this means:
+     - Merge the specified commitish
+     - Drop any cpick files included in the snapshot
+     - Refresh all quilt patches on the branch
+     - Update the changelog accordingly
+     - Tell user how to release this update
+    """
+    env = EnvDetails.get()
+    if not post_patch:
+        merge_commitish(commitish)
+        drop_cpicks(commitish)
+        refresh_patches(commitish)
+
+    # If arguments haven't been passed, we have a few things to determine
+    is_devel, first_devel_upload, first_sru = get_possible_devel_options(
+        known_first_devel_upload, known_first_sru, env
+    )
+    bug = get_sru_bug(bug, no_sru_bug)
```

Comment:
We shouldn't call this in devel series
```suggestion
    bug = None if is_devel else get_sru_bug(bug, no_sru_bug)
```